contentcuration/contentcuration/static/js/edit_channel/file_upload/views.js

```diff
@@ -566,12 +568,139 @@ var FormatSlot = BaseViews.BaseListNodeItemView.extend({
         this.set_uploading(false);
     },
     remove_item:function(){
-        this.containing_list_view.set_file_format(null, this.model, this.file);
-        this.file = null;
-        this.render();
+        if(this.language_view){
+            this.language_view.add_language(new Models.LanguageModel(this.file.get('language')));
+            this.set_file(null);
+            this.remove();
+            this.containing_list_view.render();
+        }else{
+            this.set_file(null);
+            this.render();
+        }
+
     },
     set_uploading:function(uploading){
         this.containing_list_view.set_uploading(uploading);
+    },
+    load_subfiles:function(){
+        var data = {
+            node : this.node,
+            model: this.model,
+            el: $(this.getSubdirectory()),
+            container: this.container,
+            files: this.files,
+            collection: this.collection,
+            containing_view: this,
+            content_node_view: this.content_node_view
+        }
+        this.subcontent_view = new MultiLanguageSlotList(data);
+    },
+});
+
+var MultiLanguageSlotList = FormatSlotList.extend({
+    template: require("./hbtemplates/file_upload_multilanguage_slot_list.handlebars"),
+    list_selector:">.multilanguage_list .preset_list",
```

Comment:
This is used to determine where to render the list (handled under edit_channel/views.py base class)